pandas/plotting/_matplotlib/converter.py

```diff
@@ -349,11 +348,7 @@ def __init__(self, locator, tz=None, defaultfmt: str = "%Y-%m-%d") -> None:
 class PandasAutoDateLocator(mdates.AutoDateLocator):
     def get_locator(self, dmin, dmax):
         """Pick the best locator based on a distance."""
-        delta = relativedelta(dmax, dmin)
-
-        num_days = (delta.years * 12.0 + delta.months) * 31.0 + delta.days
-        num_sec = (delta.hours * 60.0 + delta.minutes) * 60.0 + delta.seconds
-        tot_sec = num_days * 86400.0 + num_sec
+        tot_sec = (dmax - dmin).total_seconds()
```

Comment:
this looks much better, and avoids the approximations (e.g. all months have 31 days) from the original